src/handler.js

```diff
@@ -63,4 +76,12 @@ const createInternalServerErrorDetails = () => {
   return { code: httpStatus.INTERNAL_SERVER_ERROR, message: 'Your request cannot be processed at this time due to an internal server error' }
 }
 
-const createErrorDetails = (event) => isGetMethod(event) ? createNotFoundDetails(event) : createMethodNotAllowedDetails(event)
+const createBadRequestDetails = (event) => {
+  return { code: httpStatus.BAD_REQUEST, message: `Your request cannot be processed because the supplied parameter(s) ${event.queryStringParameters} cannot be understood` }
+}
+
+const createErrorDetails = (event) => {
+  return isGetMethod(event)
+    ? hasQueryParameters(event) ? createBadRequestDetails(event) : createNotFoundDetails(event)
```

Comment:
I think nesting ternary operators can become problematic. I think either it should be split using if else or refactor the line 85 to a method with name problemsWithGetMethod

Reply:
Moved to its own method. I guess it will be further broken down in the next PR's dealing with validation of query